Two new new activities, one optional parameter

ahouseholder opened the pull request:
Hi there,

I just used the h1 module to retrieve a bunch of reports out of HackerOne. It worked quite well, but I hit into three separate exceptions on different records. 

1. activity-cve-id-added
2. activity-nobody-assigned-to-bug
3. duplicate_report_id should be optional

The two activities are present in the data although they are undocumented in the H1 API docs. I also ran across at least one report object that contained an activity-external-user-joined object without the duplicate_report_id attribute, so it seems like marking it as optional is the obvious fix.

ahouseholder committed:
Add "activity-cve-id-added" and "activity-nobody-assigned-to-bug" as valid activities.

ahouseholder committed:
duplicate_report_id appears to be an optional attribute on ActivityExternalUserJoined now.

Fix for:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/adh/anaconda/envs/py36/lib/python3.6/site-packages/h1/client.py", line 109, in get_resource
    return rsrc_type.get(self, obj_id)
  File "/Users/adh/anaconda/envs/py36/lib/python3.6/site-packages/h1/models.py", line 171, in get
    return requester.request_object(cls.make_url(obj_id))
  File "/Users/adh/anaconda/envs/py36/lib/python3.6/site-packages/h1/client.py", line 85, in request_object
    return hydrate_object(data, self)
  File "/Users/adh/anaconda/envs/py36/lib/python3.6/site-packages/h1/models.py", line 47, in hydrate_object
    return hydrator(requester, obj)
  File "/Users/adh/anaconda/envs/py36/lib/python3.6/site-packages/h1/models.py", line 162, in __init__
    super(HackerOneResource, self).__init__(requester, raw_data)
  File "/Users/adh/anaconda/envs/py36/lib/python3.6/site-packages/h1/models.py", line 80, in __init__
    self._handle_data()
  File "/Users/adh/anaconda/envs/py36/lib/python3.6/site-packages/h1/models.py", line 93, in _handle_data
    self._hydrate()
  File "/Users/adh/anaconda/envs/py36/lib/python3.6/site-packages/h1/models.py", line 294, in _hydrate
    self._make_relationship("activities", self._hydrate_list_of_objects, [])
  File "/Users/adh/anaconda/envs/py36/lib/python3.6/site-packages/h1/models.py", line 128, in _make_relationship
    rel_val = hydrator(rel_val["data"])
  File "/Users/adh/anaconda/envs/py36/lib/python3.6/site-packages/h1/models.py", line 110, in _hydrate_list_of_objects
    return [self._hydrate_object(x) for x in val]
  File "/Users/adh/anaconda/envs/py36/lib/python3.6/site-packages/h1/models.py", line 110, in <listcomp>
    return [self._hydrate_object(x) for x in val]
  File "/Users/adh/anaconda/envs/py36/lib/python3.6/site-packages/h1/models.py", line 113, in _hydrate_object
    return hydrate_object(val, self._requester)
  File "/Users/adh/anaconda/envs/py36/lib/python3.6/site-packages/h1/models.py", line 47, in hydrate_object
    return hydrator(requester, obj)
  File "/Users/adh/anaconda/envs/py36/lib/python3.6/site-packages/h1/models.py", line 80, in __init__
    self._handle_data()
  File "/Users/adh/anaconda/envs/py36/lib/python3.6/site-packages/h1/models.py", line 93, in _handle_data
    self._hydrate()
  File "/Users/adh/anaconda/envs/py36/lib/python3.6/site-packages/h1/models.py", line 411, in _hydrate
    self._activity_hydrate()
  File "/Users/adh/anaconda/envs/py36/lib/python3.6/site-packages/h1/models.py", line 513, in _activity_hydrate
    self._make_attribute("duplicate_report_id", self._hydrate_verbatim)
  File "/Users/adh/anaconda/envs/py36/lib/python3.6/site-packages/h1/models.py", line 120, in _make_attribute
    attr_val = attrs[attr_name]
KeyError: 'duplicate_report_id'

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/uber-common/h1-python?pullRequest=6) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/uber-common/h1-python?pullRequest=6) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/uber-common/h1-python?pullRequest=6) it.</sub>

ahouseholder committed:
fix code style warnings

$ make lint test
flake8 h1 tests
h1/models.py:494:1: E302 expected 2 blank lines, found 1
h1/models.py:495:9: E225 missing whitespace around operator
h1/models.py:497:1: E302 expected 2 blank lines, found 1
h1/models.py:543:1: E302 expected 2 blank lines, found 1
h1/models.py:546:1: E302 expected 2 blank lines, found 1
make: *** [lint] Error 1

ahouseholder commented:
I'll resubmit with the lint warnings fixed.
